client/src/wayland_obj/output.rs

```diff
@@ -1,35 +1,52 @@
 //! Wrapper around a wl_output
 
-use std::{cell::RefCell, fmt};
+use std::{cell::RefCell, fmt, rc::Rc};
 
 use wayland_client::{
     protocol::wl_output::{self, WlOutput},
     GlobalImplementor, NewProxy, Proxy
 };
 
-use crate::area::{Area, Origin, Size};
-use crate::lua::LUA;
-use crate::objects::screen::{self, Screen};
-
 /// The minimum version of the wl_output global to bind to.
-pub const WL_OUTPUT_VERSION: u32 = 2;
+pub(crate) const WL_OUTPUT_VERSION: u32 = 2;
 
 /// Wrapper around WlOutput.
 #[derive(Clone, Eq, PartialEq)]
 pub struct Output {
     output: WlOutput
 }
 
+pub trait OutputEventHandler {
```

Comment:
Naming is a bit tricky here, `OutputEventHandler` is the trait that will be 'implemented' by the lua side, while `WlOutputEventHandler` is the implementation that is provided to the wayland side.... This should probably be documented somewhere

Reply:
Why does this trait need to exist in the first place?